src/os/bluestore/BlueStore.cc

```diff
@@ -8666,21 +8664,15 @@ void BlueStore::_deferred_submit_unlock(OpSequencer *osr)
 void BlueStore::_deferred_aio_finish(OpSequencer *osr)
 {
   dout(10) << __func__ << " osr " << osr << dendl;
-  assert(osr->deferred_running);
-  DeferredBatch *b = osr->deferred_running;
+  DeferredBatch *b;
 
   {
     std::lock_guard<std::mutex> l(deferred_lock);
-    assert(osr->deferred_running == b);
-    osr->deferred_running = nullptr;
-    if (!osr->deferred_pending) {
+    b = osr->deferred_running.front();
```

Comment:
Yeah, make sense.
But still I am afraid it won't work well as in this way `throttle_deferred_bytes.wait()` does not wake up until the throttle itself gets drained. Instead we need a mechanism to make sure we wake up each time the cost of throttle is changing..